src/templates/console.html

```diff
@@ -69,28 +70,8 @@
           }
         );
 
-        // JQuery.terminal adds a trailing newline that is not easy
-        // to remove with the current API. Doing our best here.
-        term.echo_no_newline = (function (message) {
-          // prompt can be polluted with previous echo
-          // _prompt_tail is the line part just before the prompt
-          this._prompt_tail = this._prompt_tail || '';
-          let prompt = this.get_prompt();
-          if( prompt.startsWith(this._prompt_tail) )
-            prompt = prompt.slice(this._prompt_tail.length);
-          else
-            this._prompt_tail = '';
-          // here, prompt and _prompt_tail should be correctly isolated
-          const lines = message.split('\n');
-          lines[0] = this._prompt_tail + lines[0];
-          this._prompt_tail = lines.pop();
-          this.set_prompt(this._prompt_tail + prompt);
-          if( lines.length )
-            this.echo(lines.join('\n'));
-        }).bind(term);
-
-        pyconsole.stdout_callback = term.echo_no_newline;
-        pyconsole.stderr_callback = (s) => term.error(s.replace(/\n+$/, ""));
+        pyconsole.stdout_callback = (s) => term.error(s, {newline : false});
```

Comment:
Is it OK to use the error method for the stdout though?

Reply:
Oops, typo.